lib/matplotlib/patches.py

```diff
@@ -586,16 +582,17 @@ def draw(self, renderer):
         # docstring inherited
         if not self.get_visible():
             return
-        with self._bind_draw_path_function(renderer) as draw_path:
-            path = self.get_path()
-            transform = self.get_transform()
-            tpath = transform.transform_path_non_affine(path)
-            affine = transform.get_affine()
-            draw_path(tpath, affine,
-                      # Work around a bug in the PDF and SVG renderers, which
-                      # do not draw the hatches if the facecolor is fully
-                      # transparent, but do if it is None.
-                      self._facecolor if self._facecolor[3] else None)
+        path = self.get_path()
+        transform = self.get_transform()
+        tpath = transform.transform_path_non_affine(path)
+        affine = transform.get_affine()
+        self._draw_paths_with_artist_properties(
+            renderer,
+            [(tpath, affine,
+              # Work around a bug in the PDF and SVG renderers, which
+              # do not draw the hatches if the facecolor is fully
+              # transparent, but do if it is None.
+              self._facecolor if self._facecolor[3] else None)])
```

Comment:
Optional: What do you think of introducing

```
_DrawPathArgs = namedtuple('_DrawPathArgs', 'path, transform, rgb_face')
```

The main motivation is to name the tuple elements because the function signature
where we could look up their meaning is quite far away.

With this we could write:

```suggestion
        transform = self.get_transform()
        self._draw_paths_with_artist_properties(
            renderer,
            [_DrawPathArgs(
                path=transform.transform_path_non_affine(self.get_path),
                transform=transform.get_affine(),
                # Work around a bug in the PDF and SVG renderers, which
                # do not draw the hatches if the facecolor is fully
                # transparent, but do if it is None.
                rgb_face=self._facecolor if self._facecolor[3] else None),
            ])
```